template/tsconfig.json

```diff
@@ -0,0 +1,34 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "lib": [
+      "esnext",
+      "esnext.asynciterable",
+      "dom"
+    ],
+    "esModuleInterop": true,
+    "allowJs": true,
+    "sourceMap": true,
+    "strict": true,
+    "noEmit": true,
+    "experimentalDecorators": true,
+    "baseUrl": ".",
+    "paths": {
```

Comment:
I think we need to follow the same logic as in the [jsconfig.json](https://github.com/nuxt/create-nuxt-app/blob/master/template/jsconfig.json) file:

```
    "paths": {
      <%_ if (server === 'adonis') { _%>
      "~/*": ["./resources/*"],
      "@/*": ["./resources/*"],
      <%_ } else { _%>
      "~/*": ["./*"],
      "@/*": ["./*"],
      <%_ } _%>
      "~~/*": ["./*"],
      "@@/*": ["./*"]
    }
```

Reply:
Sounds right.

It may later be refactored to prevent templates to have duplicated parts and evaluate the config values before rendering the template, that will be valid for either `jsconfig.json` / `tsconfig.json`.
For example, `compilerOptions.baseUrl`, `compilerOptions.paths` & `exclude` are identic in both templates.

Reply:
Still, if server `adonis` is selected, people may believed the `adonis` template would be written in TypeScript :/